google/cloud/spanner/client.cc

```diff
@@ -307,13 +307,22 @@ StatusOr<CommitResult> Client::Commit(Transaction transaction,
 StatusOr<CommitResult> Client::CommitAtLeastOnce(
     Transaction::ReadWriteOptions transaction_options, Mutations mutations,
     Options opts) {
+  // Note: This implementation differs from `CommitAtLeastOnce({mutations})`
+  // for historical reasons.
```

Comment:
I know of no such explanation.  Indeed, I'm only intimating that the existing `CommitAtLeastOnce(mutations)` and the new `CommitAtLeastOnce({mutations})` must behave the same (modulo the former being able to return `CommitStats`) because anything else would be crazy.  If they had thought of the new batching interface first, then I imagine the single-mutations call would never exist.